### PR TITLE
docs(api): update functions and classes available to workers

### PR DESCRIPTION
**Pull request by ChaominRuan**
### Description

Align with  https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API#web_workers_concepts_and_usage

### CODE REVIEW

File: files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md [hunk 1]
```diff
@@ -16,55 +16,66 @@ In addition to the standard [JavaScript](/en-US/docs/Web/JavaScript) set of func
...
 - {{domxref("Media Source Extensions API", "Media Source Extensions API", "", "nocode")}} (dedicated workers only)
 - [Network Information API](/en-US/docs/Web/API/Network_Information_API)
-- {{domxref("Notifications_API", "Notifications")}}
-- {{domxref("Performance")}}
-- {{domxref("PerformanceEntry")}}
-- {{domxref("PerformanceMeasure")}}
-- {{domxref("PerformanceMark")}}
-- {{domxref("PerformanceObserver")}}
-- {{domxref("PerformanceResourceTiming")}}
+- {{domxref("Notifications_API", "Notifications API")}}
```
[the remark this answers is not in the document]
Comment (Kaiido) [reply files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md:57]: We could also include {{domxref("Path2D")}} though that could also probably be part of `OffscreenCanvas`.

File: files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md [hunk 2]
```diff
@@ -16,55 +16,66 @@ In addition to the standard [JavaScript](/en-US/docs/Web/JavaScript) set of func
...
+  - {{domxref("PerformanceMeasure")}}
+  - {{domxref("PerformanceMark")}}
+  - {{domxref("PerformanceObserver")}}
+  - {{domxref("PerformanceResourceTiming")}}
 - {{jsxref("Promise")}}
 - [Server-sent events](/en-US/docs/Web/API/Server-sent_events)
 - {{domxref("ServiceWorkerRegistration")}}
-- {{domxref("TextEncoder")}} and {{domxref("TextDecoder")}}
-- {{ domxref("URL") }}
+- {{domxref("URL_API","URL API")}} (e.g. {{domxref("URL")}})
```
Comment (Kaiido) [inline files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md:68]: (+) {{domxref('WebCodecs API','','','true')}}

**Review by Kaiido — COMMENTED** [review]
Non exhaustive review as a response to @sideshowbarker call on Matrix. 

This only touches the few areas I'm aware of so it may miss some issues with the existing content and more likely, it may not have included all the missing APIs.  
And I guess you'll want to duplicate these in the second file too.
Also beware with the links I made, I wrote them by hand and didn't really check the format.

File: files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md [hunk 3]
```diff
@@ -16,55 +16,66 @@ In addition to the standard [JavaScript](/en-US/docs/Web/JavaScript) set of func
...
-Some of the functions that are common to all workers and to the main thread (from `WindowOrWorkerGlobalScope`) are: {{domxref("atob", "atob()")}}, {{domxref("btoa", "btoa()")}}, {{domxref("clearInterval", "clearInterval()")}}, {{domxref("clearTimeout()")}}, {{domxref("Window.dump()", "dump()")}} {{non-standard_inline}}, {{domxref("setInterval()")}}, {{domxref("setTimeout()")}}.
+Some of the functions (a subset) that are common to all workers and to the main thread (from `WindowOrWorkerGlobalScope`) are:
+
+- {{domxref("atob", "atob()")}}
+- {{domxref("btoa", "btoa()")}}
+- {{domxref("clearInterval", "clearInterval()")}}
+- {{domxref("clearTimeout()")}}
+- {{domxref("Window.dump()", "dump()")}} {{non-standard_inline}}
+- {{domxref("setInterval()")}}
+- {{domxref("setTimeout()")}}
```
[the remark this answers is not in the document]
Comment (Kaiido) [reply files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md:30]: Ah I missed
(+) {{domxref("queueMicrotask()")}}
(+) {{domxref("structuredClone()")}}

File: files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChaominRuan) [reply files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md:60]: OffscreenCanvas see below

File: files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md [hunk 4]
```diff
@@ -16,55 +16,66 @@ In addition to the standard [JavaScript](/en-US/docs/Web/JavaScript) set of func
...
-- {{domxref("PerformanceResourceTiming")}}
+- {{domxref("Notifications_API", "Notifications API")}}
+- {{domxref("Performance_API","Performance API")}}, including:
+  - {{domxref("Performance")}}
+  - {{domxref("PerformanceEntry")}}
+  - {{domxref("PerformanceMeasure")}}
+  - {{domxref("PerformanceMark")}}
+  - {{domxref("PerformanceObserver")}}
+  - {{domxref("PerformanceResourceTiming")}}
 - {{jsxref("Promise")}}
```
[the remark this answers is not in the document]
Comment (ChaominRuan) [reply files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md]: Yes, it has been deleted

File: files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md [hunk 5]
```diff
@@ -16,55 +16,66 @@ In addition to the standard [JavaScript](/en-US/docs/Web/JavaScript) set of func
...
 - [Server-sent events](/en-US/docs/Web/API/Server-sent_events)
 - {{domxref("ServiceWorkerRegistration")}}
-- {{domxref("TextEncoder")}} and {{domxref("TextDecoder")}}
-- {{ domxref("URL") }}
+- {{domxref("URL_API","URL API")}} (e.g. {{domxref("URL")}})
 - [WebGL](/en-US/docs/Web/API/WebGL_API) with {{domxref("OffscreenCanvas")}}
```
[the remark this answers is not in the document]
Comment (sideshowbarker) [reply files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md:74]: @ChaominRuan This comment still need to be addressed, right?

File: files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (sideshowbarker) [reply files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md:62]: @ChaominRuan This comment still need to be addressed, right?

File: files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md [hunk 6]
```diff
@@ -16,55 +16,71 @@ In addition to the standard [JavaScript](/en-US/docs/Web/JavaScript) set of func
...
+- {{domxref("atob", "atob()")}}
+- {{domxref("btoa", "btoa()")}}
+- {{domxref("clearInterval", "clearInterval()")}}
+- {{domxref("clearTimeout()")}}
+- {{domxref("Window.dump()", "dump()")}} {{non-standard_inline}}
+- {{domxref("setInterval()")}}
+- {{domxref("setTimeout()")}}
+- {{domxref("queueMicrotask()")}}
+- {{domxref("structuredClone()")}}
+- {{domxref("window.requestAnimationFrame")}}
```
Comment (Kaiido) [inline files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md:30]: This one too is (dedicated workers only)

File: files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md [hunk 7]
```diff
@@ -16,55 +16,71 @@ In addition to the standard [JavaScript](/en-US/docs/Web/JavaScript) set of func
...
+- [Web Crypto API](/en-US/docs/Web/API/Web_Crypto_API) (e.g. {{domxref("Crypto")}})
+- [CSS Font Loading API](/en-US/docs/Web/API/CSS_Font_Loading_API)
+- {{domxref("CustomEvent")}}
+- {{domxref("Encoding_API", "Encoding API")}} (e.g. {{domxref("TextEncoder")}}, {{domxref("TextDecoder")}})
+- {{domxref("Fetch_API", "Fetch API")}}
 - {{domxref("FileReader")}}
 - {{domxref("FileReaderSync")}} (only works in workers!)
 - {{domxref("FormData")}}
 - {{domxref("ImageData")}}
+- {{domxref("ImageBitmap")}}
```
Comment (Kaiido) [inline files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md:58]: If we follow the alphabetical order I think this should go before `ImageData`.

File: files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md [hunk 8]
```diff
@@ -16,55 +16,71 @@ In addition to the standard [JavaScript](/en-US/docs/Web/JavaScript) set of func
...
 - [Network Information API](/en-US/docs/Web/API/Network_Information_API)
-- {{domxref("Notifications_API", "Notifications")}}
-- {{domxref("Performance")}}
-- {{domxref("PerformanceEntry")}}
-- {{domxref("PerformanceMeasure")}}
-- {{domxref("PerformanceMark")}}
-- {{domxref("PerformanceObserver")}}
-- {{domxref("PerformanceResourceTiming")}}
-- {{jsxref("Promise")}}
+- {{domxref("Notifications_API", "Notifications API")}}
```
Comment (Kaiido) [inline files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md:62]: (+) {{domxref("OffscreenCanvas")}} (and all the canvas context APIs)

File: files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md [hunk 9]
```diff
@@ -16,55 +16,71 @@ In addition to the standard [JavaScript](/en-US/docs/Web/JavaScript) set of func
...
+  - {{domxref("PerformanceMark")}}
+  - {{domxref("PerformanceObserver")}}
+  - {{domxref("PerformanceResourceTiming")}}
 - [Server-sent events](/en-US/docs/Web/API/Server-sent_events)
 - {{domxref("ServiceWorkerRegistration")}}
-- {{domxref("TextEncoder")}} and {{domxref("TextDecoder")}}
-- {{ domxref("URL") }}
+- {{domxref("URL_API","URL API")}} (e.g. {{domxref("URL")}})
+- {{domxref('WebCodecs_API','','','true')}}
 - [WebGL](/en-US/docs/Web/API/WebGL_API) with {{domxref("OffscreenCanvas")}}
```
Comment (Kaiido) [inline files/en-us/web/api/web_workers_api/functions_and_classes_available_to_workers/index.md:74]: I think we should remove WebGL.

**Review by Kaiido — COMMENTED** [review]
I think it would still be great to have yet another pair of eyes on this.

**Review by sideshowbarker — APPROVED** [review]

**Comment by sideshowbarker** [comment]
> I think it would still be great to have yet another pair of eyes on this.

I did a review as well, so I think we’re good to go.

**Comment by sideshowbarker** [comment]
@ChaominRuan, thanks much, and congrats on landing your first docs change here — welcome aboard 🎉

And thanks much @Kaiido for the detailed review.

**Comment by ChaominRuan** [comment]
@sideshowbarker ,Thanks for the merged, And thanks much @Kaiido for the detailed review.

**Pull request merged**